src/Compilers/CSharp/Portable/Symbols/Source/TypeParameterConstraintClause.cs

```diff
@@ -73,11 +84,11 @@ internal sealed class TypeParameterConstraintClause
 
         internal bool IsEmpty => Constraints == TypeParameterConstraintKind.None && ConstraintTypes.IsEmpty;
 
-        internal bool IsEarly => !Syntax.IsDefault;
+        internal bool IsEarly => !TypeConstraintsSyntax.IsDefault;
```

Comment:
>!TypeConstraintsSyntax.IsDefault [](start = 33, length = 32)

Minor point: perhaps change the definition to `ClauseSyntax != null` since that feels a little more natural.